packages/massa-web3/src/experimental/smartContract.ts

```diff
@@ -106,24 +47,27 @@ function populateDatastore(
   const datastore = new Map<Uint8Array, Uint8Array>()
 
   // set the number of contracts in the first key of the datastore
-  datastore.set(new Uint8Array([0x00]), u64ToBytes(BigInt(contracts.length)))
+  datastore.set(
+    new Uint8Array([MASTER_KEY]),
+    u64ToBytes(BigInt(contracts.length))
+  )
 
   contracts.forEach((contract, i) => {
-    datastore.set(u64ToBytes(BigInt(i + 1)), contract.data)
+    datastore.set(u64ToBytes(BigInt(i + ONE)), contract.data)
     if (contract.args) {
       datastore.set(
         new Args()
-          .addU64(BigInt(i + 1))
-          .addUint8Array(u8toByte(0))
+          .addU64(BigInt(i + ONE))
+          .addUint8Array(u8toByte(ZERO))
           .serialize(),
```

Comment:
To improve code raedibility, you should add a function named argumentKey(offset: number) that returned the key. Because, right now, I need to think to understand what this code is doing.